modules/internal/ChapelArray.chpl

```diff
@@ -1558,37 +1558,59 @@ module ChapelArray {
     /* Return true if this domain is a subset of ``super``. Otherwise
        returns false. */
     proc isSubset(super : domain) {
-      if !isAssociativeDom(this) {
-        if isRectangularDom(this) then
-          compilerError("isSubset not supported on rectangular domains");
-        else if isOpaqueDom(this) then
+      if !(isAssociativeDom(this) || isRectangularDom(this)) {
+        if isOpaqueDom(this) then
           compilerError("isSubset not supported on opaque domains");
         else if isSparseDom(this) then
           compilerError("isSubset not supported on sparse domains");
         else
           compilerError("isSubset not supported on this domain type");
       }
       if super.type != this.type then
-        compilerError("isSubset called with different associative domain types");
+        if isRectangularDom(this) {
+          if super.dims().size != this.dims().size then
+            compilerError("isSubset called with different rectangular domain types");
```

Comment:
I think this should say something about the ranks of the domains being different, not just 'types'. For example: "rank mismatch in domain.isSubset".

I think there should also be a check to ensure the index types are the same.

You could also use 'super.rank != this.rank' instead of checking the dims size.

Reply:
I see. I didn't know about `rank`, thanks for pointing that out!

I'll update the error.

About the index types, currently Chapel only supports domains containing ranges with integral bounds, AFAIK. Would it still be necessary to check the index types?

Reply:
There are lots of integral types, e.g. `int(8)`, `uint(32)`, `int`, `uint`.

Reply:
> There are lots of integral types, e.g. `int(8)`, `uint(32)`, `int`, `uint`.

Understood. Also, I'm not sure how to check the index type in a range. Could you please point me to some documentation here?

Reply:
someRange.low.type and then you can compare that type to another with ==